sdks/python/apache_beam/runners/worker/sdk_worker_main.py

```diff
@@ -175,20 +175,32 @@ def create_harness(environment, dry_run=False):
 
 def main(unused_argv):
   """Main entry point for SDK Fn Harness."""
-  fn_log_handler, sdk_harness, sdk_pipeline_options = create_harness(os.environ)
-  experiments = sdk_pipeline_options.view_as(DebugOptions).experiments or []
+  (fn_log_handler, sdk_harness,
+   sdk_pipeline_options) = create_harness(os.environ)
   dataflow_service_options = (
       sdk_pipeline_options.view_as(GoogleCloudOptions).dataflow_service_options
       or [])
-  if (_ENABLE_GOOGLE_CLOUD_PROFILER in experiments) or (
-      _ENABLE_GOOGLE_CLOUD_PROFILER in dataflow_service_options):
+  experiments = (sdk_pipeline_options.view_as(DebugOptions).experiments or [])
```

Comment:
I think it would be sufficient to only parse Dataflow service options, since enable_google_cloud_profiler is advertized as a service option.

Reply:
fixed